test/cmdlineTests.sh

```diff
@@ -530,6 +585,36 @@ printTask "Testing assemble, yul, strict-assembly and optimize..."
     test_solc_assembly_output "{ let x := 0 }" "{ { } }" "--strict-assembly --optimize"
 )
 
+printTask "Testing the eqivalence of --experimental-via-ir equivalence and a two-stage compilation..."
+(
+    printTask " - Smoke test"
+    test_via_ir_equivalence "contract C {}"
+
+    printTask " - Smoke test (optimized)"
+    test_via_ir_equivalence "contract C {}" --optimize
+
+    externalContracts=(
+        deposit_contract.sol
+        FixedFeeRegistrar.sol
+        _stringutils/stringutils.sol
```

Comment:
I'm actually more worried about there not being multi-file or multi-contract tests.

Reply:
That's doable for multi-file but I'd have to restructure the script a bit and pass some extra paths as arguments. It would be easiest with a map and that's a real pain in Bash so I'd probably just do it in Python. Should I do that? We probably have enough single-file contracts that I didn't think it was worth spending time on.

For multi-contract it's because in Yul mode we only support compiling one file at a time so I'd have to manually split the IR output, assemble and then combine again in the right order before diffing. If we really want that, I'd rather just add a multi-file mode in the CLI than cobble it up in Bash.